Justfile

```diff
@@ -0,0 +1,176 @@
+        unzip -o {{ cache_dir }}/{{ godot_filename }}.zip -d {{ cache_dir }}
+
+        if [ "{{ os() }}" = "macos" ]; then
+            cp {{ cache_dir }}/Godot.app/Contents/MacOS/Godot {{ godot_bin }}
+        else
+            cp {{ cache_dir }}/{{ godot_filename }} {{ godot_bin }}
+        fi
+    fi
+
+# Download plugins
+install-addons:
+    [ -f plug.gd ] && just godot --no-window -s plug.gd install || true
+
+# Import game resources
+import-resources:
+    just godot --no-window --editor --quit
+
+# Updates the addon version
+@bump-version:
+    echo "Update version in the plugin.cfg"
+    sed -i "s,version=.*$,version=\"{{ addon_version }}\",g" ./addons/{{ addon_name }}/plugin.cfg
+    echo "Update version in the README.md"
+    sed -i "s,tag = ".*"$,tag = \"{{ addon_version }}\"\, include = [\"addons/glam\"]}),g" ./README.md
+
+# Godot binary wrapper
+@godot *ARGS: makedirs install-godot
+    #!/usr/bin/env sh
+    if [ "{{ use_x11_wrapper }}" = "true" ]; then
+        just ci-godot-x11 {{ ARGS }}
+    else
+        {{ godot_bin }} {{ ARGS }}
+    fi
+
+# Open the Godot editor
+editor:
+    just godot --editor
+
+# Run files formatters
+fmt:
+    just venv pip install pre-commit==3.5.0 reuse==2.1.0 gdtoolkit==3.*
+    just venv pre-commit run -a
+
+# Remove cache and binaries created by this Justfile
+[private]
+clean-glam-cache:
+    rm -rf {{ main_dir }}
+    rm -rf {{ venv_dir }}
+
+# Remove plugins
+clean-addons:
+    rm -rf .plugged
+    [ -f plug.gd ] && find addons/ -maxdepth 1 -type d -not -name 'addons' -not -name 'gd-plug' -not -name '{{ addon_name }}' -exec rm -rf {} \; || true
+
+# Remove any unnecessary files
+clean: clean-addons
+
+# Add some variables to Github env
+ci-load-dotenv:
+    echo "godot_version={{ godot_version }}" >> $GITHUB_ENV
+    echo "addon_name={{ addon_name }}" >> $GITHUB_ENV
+    echo "addon_version={{ addon_version }}" >> $GITHUB_ENV
+
+# Starts godot using Xvfb and pulseaudio
+ci-godot-x11 *ARGS:
+    #!/bin/bash
+    set -e
+    # Set locale to 'en' if locale is not already set.
+    # Godot will fallback to this locale anyway and it
+    # prevents an error message being printed to console.
+    [ "$LANG" == "C.UTF-8" ] && LANG=en || true
+
+    # Start dummy sound device.
+    pulseaudio --check || pulseaudio -D
+
+    # Running godot with X11 Display.
+    xvfb-run --auto-servernum {{ godot_bin }} {{ ARGS }}
+
+    # Cleanup (allowed to fail).
+    pulseaudio -k || true
+
+# Upload the addon on Github
+publish:
+    gh release create "{{ addon_version }}" --title="v{{ addon_version }}" --generate-notes
+    # TODO: Add an asset-lib publish step
+
+# Run unit tests
+unit: install-addons import-resources
+    just godot --no-window -s addons/gut/gut_cmdln.gd -gconfig=.gutconfig.json
+
+# Run integration tests
+integration: install-addons import-resources
+    just godot --no-window -s addons/gut/gut_cmdln.gd -gdir=res://test/integration/sources -gexit
```

Comment:
**issue:** This is only running some of the integration tests. To run all, it should be:
```suggestion
    just godot --no-window -s addons/gut/gut_cmdln.gd -gdir=res://test/integration -gexit
```